### PR TITLE
Always throw if unable to overwrite

### LINKED ISSUE #37: need feedback if copy fails because overwrite: false
When using `overwrite: false`, there's no feedback (apart from a progress bar, perhaps, but that's not relevant to my use case) as to whether or not the copy succeeded. this error is _explicitly_ eaten, so I realize rejecting here is _not_ the desired behavior.

But I'm wondering if it's possible to provide _some_ sort of feedback in this case?  Either resolve with a value, or add another option to enable reject-on-existing-file...

Please let me know what you think; I can probably send a PR.

### PR DESCRIPTION
**Pull request by boneskull**
This is a breaking change

Fixes #37

### CODE REVIEW

File: test/async.js [hunk 1]
```diff
@@ -7,9 +7,9 @@ import del from 'del';
 import test from 'ava';
 import uuid from 'uuid';
 import sinon from 'sinon';
-import cpFile from '..';
 import assertDateEqual from './helpers/_assert';
 import {buildEACCES, buildEIO, buildENOSPC, buildENOENT, buildEPERM} from './helpers/_fs-errors';
+import cpFile from '..';
```
Comment (boneskull) [inline test/async.js]: this was from `xo` error

File: test/sync.js [hunk 2]
```diff
@@ -5,9 +5,9 @@ import del from 'del';
 import test from 'ava';
 import uuid from 'uuid';
 import sinon from 'sinon';
-import cpFile from '..';
 import assertDateEqual from './helpers/_assert';
 import {buildEACCES, buildENOSPC, buildEBADF, buildEPERM} from './helpers/_fs-errors';
+import cpFile from '..';
```
Comment (boneskull) [inline test/sync.js]: `xo` again

**Comment by boneskull** [comment]
Maybe more appropriate to say "always throw (or reject) on copy failure"

**Comment by boneskull** [comment]
node v12.13.0 introduced a breaking change, I'll investigate

**Comment by boneskull** [comment]
I've added a change to work around the issue (version sniffing! fun!)

**Comment by boneskull** [comment]
@sindresorhus You may want to comment on nodejs/node#29930 as I don't think I can be very much help

**Comment by mcollina** [comment]
I think you should wait for `open` to be emitted on the writable stream as well before piping. I’m not sure why this was not causing problem _before_, as it seems a race condition to me between the readable and writable sides.

**Comment by boneskull** [comment]
I'm not sure what's going on here.

**Comment by sindresorhus** [comment]
@boneskull It's not clear to me what the status of this PR is?

**Comment by boneskull** [comment]
I’m not sure what to do with @matteocollina’s feedback. there’s a workaround in this PR, and afaik the PR _works_, but I’m unsure if I should be changing something else per his feedback

**Comment by sindresorhus** [comment]
@mcollina The docs says nothing about having to wait for an `open` event: https://nodejs.org/api/fs.html#fs_fs_createwritestream_path_options

**Comment by sindresorhus** [comment]
I've added it just in case though: https://github.com/sindresorhus/cp-file/commit/6d92d5cafc51bddd5b2bf1b49f2393e3e9357191

**Comment by sindresorhus** [comment]
Reverted that ^. It caused problems on Node.js 12 (I apparently had an older 12 version locally so I didn't detect the problem before pushing). https://travis-ci.org/sindresorhus/cp-file/jobs/654317803?utm_medium=notification&utm_source=email

**Pull request merged**